src/protocols/memory_attributes.rs

```diff
@@ -0,0 +1,40 @@
+//! Memory Attributes Protocol
```

Comment:
Spec uses singular `attribute`:

```suggestion
//! Memory Attribute Protocol
```

Can you also adjust the file-name to `memory_attribute.rs`?

Reply:
Updated both.